src/lib.rs

```diff
@@ -177,8 +177,12 @@ impl IpfsOptions {
     ///
     /// Also used from examples.
     pub fn inmemory_with_generated_keys() -> Self {
+        use tempfile::TempDir;
+
+        let tempdir = TempDir::new().expect("tempdir creation failed").into_path();
+
         Self {
-            ipfs_path: env::temp_dir(),
+            ipfs_path: tempdir,
```

Comment:
does this mean that the dev system gets polluted with temporary directories?

Reply:
This should only be using when testing and running the examples, as far as I understand, the directory is deleted when the temp dir goes out of scope. It could be explicitly done with `TempDir::close` though not sure where that would have to go `repo::shutdown` perhaps? I believe tests rely on drop at the moment (no explicit call to shutdown afaik), so this would need changing elsewhere as well. I'll investigate 👍

Reply:
You were 💯 on the money there, so I tried something else, not sure if it is the right approach but it seems to work. This only creates a lock if the repo is on disk, the lock struct is empty if in memory (which is what we use for tests). I've checked that this implementation doesn't leave any temp dirs.